samples/spring-eventsourced-customer-registry/src/main/java/customer/view/CustomerView.java

```diff
@@ -35,25 +25,20 @@ public static Optional<CustomerView> onEvent(CustomerView state, CustomerEvent e
 
     if (state == null) {
```

Comment:
@octonato how about changing this to sth like:
```
public static Optional<CustomerView> onEvent(CustomerView state, CustomerEvent event) {

    // the only event we can receive when state is null is the CustomerCreated
    if (state == null && event instanceof CustomerCreated created) {
      return Optional.of(new CustomerView(created.email(), created.name(), created.address()));
    } else if (event instanceof NameChanged nameChanged) {
      return Optional.of(state.withName(nameChanged.newName()));
    } else if (event instanceof AddressChanged addressChanged) {
      return Optional.of(state.withAddress(addressChanged.address()));
    } else {
      throw new IllegalStateException("Not handled event: " + event);
    }
  }
```

Reply:
sth? You mean switch statement? 

Francisco's PR on event handler should have solved this. We just forgot to update the samples. It should be possible now to have overloaded `onEvent` methods receiving each a concrete type. Or simply different method names. Doesn't matter much.

Reply:
ok